resources/definition.rb

```diff
@@ -8,19 +13,32 @@
   default: '/usr/local/ruby',
   description: 'Location to install Ruby'
 
-property :environment, String,
-  description: 'Environment to pass to the ruby-build install process'
+# NOTE: adding the Ruby version to the installation prefix
+# by default is unexpected and will likely lead to user
+# problems. Now defaults to false.
+#
+property :version_prefix, [true, false],
+  default: false,
+  description: 'add Ruby version to the installation prefix'
+
+property :patch, [String, nil],
+  description: 'path to a Ruby patch file for ruby-build to use'
+
+property :environment, Hash,
+  default: {},
+  description: 'Environment hash to pass to the ruby-build install process'
 
 property :user, String,
   description: 'User to install as'
 
 property :group, String,
   description: 'Group to install as'
 
-unified_mode true
+unified_mode true if respond_to? :unified_mode
 
 action :install do
-  Chef::Log.fatal 'JRuby is not a supported definition' if new_resource.definition.include? 'jruby'
+  Chef::Log.fatal('JRuby is not a supported definition') \
```

Comment:
what's wrong with the line length here?

Reply:
It's me being picky, and maybe too picky. Long line length is generally frowned on, but it can be hard to keep that going with the Chef DSL.

Reply:
(To be specific: this line is almost 100 characters long, and idiomatic Ruby wants to be < 80 characters. Like I said, this is just me being a little picky, and it's not really necessary. Let me know if you want me to undo this.)

Reply:
Meh I’m really not bothered by it either way. Cookstyle favours readability over rubocops archaic 80 character length.